pkg/backup/backupschedule/backup_schedule_manager.go

```diff
@@ -167,6 +292,51 @@ func (bm *backupScheduleManager) canPerformNextBackup(bs *v1alpha1.BackupSchedul
 	return nil
 }
 
+// canPerformNextCompact handles the compact backup processing logic.
+// It returns a controller.RequeueError if the backup is still running,
+// otherwise it updates the LastCompactTime or returns any encountered error.
+func (bm *backupScheduleManager) canPerformNextCompact(bs *v1alpha1.BackupSchedule) error {
+	if bs.Status.LastCompact == "" {
+		return nil
+	}
+
+	lastCompact := bs.Status.LastCompact
+	ns := bs.GetNamespace()
+	bsName := bs.GetName()
+
+	compact, err := bm.deps.CompactBackupLister.CompactBackups(ns).Get(lastCompact)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return nil
+		}
+		return fmt.Errorf("backup schedule %s/%s: get compact backup %s failed: %w", ns, bsName, lastCompact, err)
+	}
+
+	var lastCompactTs string
+	switch compact.Status.State {
+	case string(v1alpha1.BackupComplete):
+		lastCompactTs = compact.Spec.EndTs
+	case string(v1alpha1.BackupFailed):
+		lastCompactTs = compact.Spec.StartTs
+	default:
+		return controller.RequeueErrorf("backup schedule %s/%s: compact backup %s is still running", ns, bsName, lastCompact)
+	}
+
+	lastCompactTime, err := config.ParseTSStringToGoTime(lastCompactTs)
+	if err != nil {
+		return perrors.AddStack(err)
+	}
+
+	if !lastCompactTime.IsZero() && lastCompactTime.Before(bs.Status.LogBackupStartTs.Time) {
```

Comment:
If the two following judgement is `false`, all the compact can't be scheduled. Is it as expected? I think `lastCompactTime` before log start time or last compact time in status is not a blocked problem.